src/lib.rs

```diff
@@ -104,6 +104,23 @@ impl<T> Vec<T> {
         }
     }
 
+    /// Creates a new `Vec<T>` reusing allocated space from the `self`.
+    ///
+    /// # Examples
+    /// ```
+    /// let vec = boxcar::Vec::new();
+    /// vec.push(1);
+    /// vec.push(2);
+    ///
+    /// let mut new_vec = vec.reset();
+    /// assert_eq!(new_vec.count(), 0);
+    /// new_vec.push(3); // will not allocate
+    /// ```
+    pub fn reset(mut self) -> Self {
```

Comment:
Can we rename this to `clear` to be consistent with `std`?

Reply:
The problem is that all std's `clear` are accepting `&mut self`. I would found it really misleading for current fn signature

Reply:
Hmm why doesn't this take `&mut self`? I missed that while reviewing.

Reply:
Well, then the property "append-only" vec is no longer held apparently. It certainly could be `&mut self`, and that's perhaps more useful function, yet it contradicts the crate definition as the elements could be removed. For now it could be seen as a "move constructor".

Reply:
I don't have a problem with making it `&mut self`, I don't think we should stick to an arbitrary definition to limit the usefulness of the type.